app/uk/gov/hmrc/economiccrimelevyregistration/controllers/CheckYourAnswersController.scala

```diff
@@ -93,21 +94,33 @@ class CheckYourAnswersController @Inject() (
 
     val base64EncodedHtmlView: String = base64EncodeHtmlView(htmlView.body)
 
+    val entityType = request.registration.entityType
+
     for {
       _        <- eclRegistrationConnector.upsertRegistration(registration =
                     request.registration.copy(base64EncodedNrsSubmissionHtml = Some(base64EncodedHtmlView))
                   )
-      response <- eclRegistrationConnector.submitRegistration(request.internalId)
-      _         = emailService.sendRegistrationSubmittedEmails(request.registration.contacts, response.eclReference)
+      response <- eclRegistrationConnector.submitRegistration(request.internalId, entityType)
```

Comment:
As per my comment about the connector - I would introduce some optionality here for now, if it is Other entity type then you don't expect a `CreateEclSubscriptionResponse` .. so this can be `Option[...]`

Reply:
See above.